docs/naps/7-key-binding-dispatch.md

```diff
@@ -0,0 +1,427 @@
+from dataclasses import dataclass, field
+
+from app_model.expressions import Expr
+
+@dataclass(order=True)
+class KeyBindingEntry:
+    command_id: str = field(compare=False)
+    weight: int
+    when: Optional[Expr] = field(compare=False)
+    block_rule: bool = field(init=False)
+    negate_rule: bool = field(init=False)
+    
+    def __post_init__(self):
+        self.block_rule = self.command_id == ''
+        self.negate_rule = self.command_id.startswith('-')
+```
+
+### Direct conflicts
+
+When two key bindings share the same key sequence, they are considered to be in direct conflict. They are sorted first according to their weight, then whether they are a blocking rule, whether they are a negate rule, and otherwise, based on their insertion order. This is done in ascending order such that higher weights and blocking/negate rules are moved further down the list.
+
+Key bindings will automatically be assigned weights depending on who set them, prioritizing default ones the least and user-set ones the most:
+
+```python
+from enum import IntEnum
+
+class KeyBindingWeights(IntEnum):
+    CORE = 0
+    PLUGIN = 300
+    USER = 500
+```
+
+### Indirect conflicts
+
+When a key sequence matches a key binding and is also a sub-sequence of a key sequence used by another key binding, it is considered an indirect conflict.
+
+There are two ways indirect conflicts can exist:
+
+A. the provided key sequence is a single modifier that is a modifier in another key binding's key combination or is a modifier in the first key combination of a key binding's key chord
+
+B. the provided key sequence is a base key or key combination that is the first part of another key binding's key chord
+
+In case (A), the corresponding command will not be triggered immediately, but will be delayed by user-defined miliseconds (e.g. 200ms), after which the press logic for the command will execute. If another key binding is triggered, this action will be canceled. If the base key is released early, the press logic will execute immediately and the delayed action will be canceled, along with the release logic being executed immediately afterwards.
+
+In case (B), the corresponding command will never be triggered so long as it indirectly conflicts with another key binding. In this sense, multi-part key bindings will always take priority over single-part key bindings.
+
+### Finding a match
+
+When checking if an active key binding matches the entered key sequence, the resolver will fetch the pre-sorted list of direct conflicts and check if the last entry is active using its `when` property, moving to the next entry if it is not. When it encounters a blocking rule, it will return no match, and for a negate rule, it will store the affected command in an ignore list and continue to the next entry. If no special rules are present, it will return a match if the command is not in an ignore list, otherwise continuing to the next entry, and so on, until no more entries remain.
```

Comment:
I don't know what blocking or negate rules are... Could you add a paragraph on these?